### PR TITLE
lc 104, lc111, levelOrderSuccessor

### PR DESCRIPTION
**Pull request by hetobias**
lc 104, lc111, levelOrderSuccessor, updated README

### CODE REVIEW

File: tobiasHe/assignments/treesBFS/lc104/lc104.js [hunk 1]
```diff
@@ -3,19 +3,33 @@
...
 
-
 function find_maximum_depth(root) {
-  //Write Code Here
-  return -1
-}
+	//Write Code Here
+	//traverse tree using BFS
+	//increment counter at each level
+	//return counter;
```
Comment (knrodriguez) [inline tobiasHe/assignments/treesBFS/lc104/lc104.js:17]: It would be cleaner to write this approach before the function declaration.

File: tobiasHe/assignments/treesBFS/lc104/lc104.js [hunk 2]
```diff
@@ -3,19 +3,33 @@
...
-    this.val = val;
-    this.left = null;
-    this.right = null;
-  }
+	constructor(val) {
+		this.val = val;
+		this.left = null;
+		this.right = null;
+	}
 }
```
Comment (knrodriguez) [inline tobiasHe/assignments/treesBFS/lc104/lc104.js:12]: Missing edge cases

File: tobiasHe/assignments/treesBFS/lc104/lc104.js [hunk 3]
```diff
@@ -3,19 +3,33 @@
...
+		let queueLength = queue.length;
+		for (let i = 0; i < queueLength; i++) {
+			let current = queue.shift();
+			if (current.left) queue.push(current.left);
+			if (current.right) queue.push(current.right);
+		}
+	}
 
+	return depthCounter;
+}
```
Comment (knrodriguez) [inline tobiasHe/assignments/treesBFS/lc104/lc104.js:32]: Don't forget to write the solution twice to build your muscle memory!

File: tobiasHe/assignments/treesBFS/lc104/lc104.js [hunk 4]
```diff
@@ -3,19 +3,33 @@
...
```
Comment (knrodriguez) [inline tobiasHe/assignments/treesBFS/lc104/lc104.js:33]: Missing test case to match edge cases.

File: tobiasHe/assignments/treesBFS/lc111/lc111.js [hunk 5]
```diff
@@ -1,30 +1,56 @@
...
 
 const find_minimum_depth = function(root) {
-  // TODO: Write your code here
-  return -1;
+	// TODO: Write your code here
+	/* 
+    need a counter for levels
+    traverse using BFS
+    start counter at 1 
+    after every level (outside of for loop) we increment counter
```
Comment (knrodriguez) [inline tobiasHe/assignments/treesBFS/lc111/lc111.js:26]: Would be cleaner if this approach was above the function.

File: tobiasHe/assignments/treesBFS/lc111/lc111.js [hunk 6]
```diff
@@ -1,30 +1,56 @@
...
+				return depthCounter;
+			} else {
+				if (current.left) queue.push(current.left);
+				if (current.right) queue.push(current.right);
+			}
+		}
+		depthCounter++;
+	}
+
+	return depthCounter;
```
Comment (knrodriguez) [inline tobiasHe/assignments/treesBFS/lc111/lc111.js:45]: Don't forget to write this twice to build muscle memory!

File: tobiasHe/assignments/treesBFS/lc111/lc111.js [hunk 7]
```diff
@@ -1,30 +1,56 @@
...
 };
```
Comment (knrodriguez) [inline tobiasHe/assignments/treesBFS/lc111/lc111.js:47]: Missing test case to match edge case.

File: tobiasHe/assignments/treesBFS/levelOrderSuccessor/levelOrderSuccessor.js [hunk 8]
```diff
@@ -22,31 +21,41 @@
...
+	}
+}
 
 const find_successor = function(root, key) {
-  // TODO: Write your code here
-  return null;
+	// TODO: Write your code here
+	// traverse tree using BFS.
+	// if key===currentNode
+	//  return the next node
```
Comment (knrodriguez) [inline tobiasHe/assignments/treesBFS/levelOrderSuccessor/levelOrderSuccessor.js:35]: Would be cleaner if the approach was above the function declaration.

File: tobiasHe/assignments/treesBFS/levelOrderSuccessor/levelOrderSuccessor.js [hunk 9]
```diff
@@ -22,31 +21,41 @@
...
+			let currentNode = queue.shift();
+			if (currentNode.left) queue.push(currentNode.left);
+			if (currentNode.right) queue.push(currentNode.right);
+			if (currentNode.val === key) {
+				return queue.shift();
+			}
+		}
+	}
+	return null;
 };
```
Comment (knrodriguez) [inline tobiasHe/assignments/treesBFS/levelOrderSuccessor/levelOrderSuccessor.js:50]: Don't forget to write it twice to build up your muscle memory!

File: tobiasHe/assignments/treesBFS/levelOrderSuccessor/levelOrderSuccessor.js [hunk 10]
```diff
@@ -22,31 +21,41 @@
...
```
Comment (knrodriguez) [inline tobiasHe/assignments/treesBFS/levelOrderSuccessor/levelOrderSuccessor.js:51]: Missing test case to match edge case.

**Review by knrodriguez — APPROVED** [review]
